tests/queries/test_qs_combinators.py

```diff
@@ -88,6 +88,10 @@ def test_union_none_slice(self):
         qs3 = qs1.union(qs2)
         self.assertNumbersEqual(qs3[:1], [0])
 
+    def test_empty_union_slice(self):
+        qs = Number.objects.union()
+        self.assertNumbersEqual(qs[:1], [0])
```

Comment:
As explained above that won't do what you think it does and could break in unexpected ways. The way things are setup right now this results in the following SQL

```sql
SELECT "queries_number"."id" AS "col1",
       "queries_number"."num" AS "col2",
       "queries_number"."other_num" AS "col3",
       "queries_number"."another_num" AS "col4"
FROM "queries_number"
UNION
SELECT "queries_number"."id" AS "col1",
       "queries_number"."num" AS "col2",
       "queries_number"."other_num" AS "col3",
       "queries_number"."another_num" AS "col4"
FROM "queries_number"
```

when it should be

```sql
SELECT "queries_number"."id" AS "col1",
       "queries_number"."num" AS "col2",
       "queries_number"."other_num" AS "col3",
       "queries_number"."another_num" AS "col4"
FROM "queries_number"
```

your test will fail with the following changes

```suggestion
    def test_empty_union_slice(self):
        qs = Number.objects.union(all=True)
        self.assertNumbersEqual(qs[:1], [0])
        self.assertNumbersEqual(qs.order_by("num")[1:], list(range(0, 10)))
```